src/Message.cpp

```diff
@@ -110,6 +179,174 @@ void MessageManager::Initialize()
 			}
 		});
 	});
+
+	Network::Get()->WebSocket().RegisterEvent(WebSocket::Event::MESSAGE_REACTION_ADD, [](json const& data)
+	{
+		Snowflake_t user_id, message_id;
+		if (!utils::TryGetJsonValue(data, user_id, "user_id"))
+			return;
+
+		if (!utils::TryGetJsonValue(data, message_id, "message_id"))
+			return;
+
+		PawnDispatcher::Get()->Dispatch([data, user_id, message_id]() mutable
+		{
+			auto const& msg = MessageManager::Get()->FindById(message_id);
+			auto const& user = UserManager::Get()->FindUserById(user_id);
+			if (msg && user)
+			{
+				Snowflake_t emoji_id;
+				std::string name;
+				utils::TryGetJsonValue(data["emoji"], emoji_id, "id");
+				if (!utils::TryGetJsonValue(data["emoji"], name, "name"))
+					return;
```

Comment:
You should move the JSON parsing out of the PAWN dispatcher lambda, as everything within it will be executed in the PAWN thread (while everything within the `RegisterEvent` lambda will run in the WebSocket network thread). This should perform a little bit better, and you won't have to pass/copy the whole `data` JSON object into the function.

You can also use
```cpp
utils::TryGetJsonValue(data, emoji_id, "emoji", "id");
```
instead of 
```cpp
utils::TryGetJsonValue(data["emoji"], emoji_id, "id");
```